Library/Homebrew/brew.rb

```diff
@@ -28,7 +21,12 @@
 class MissingEnvironmentVariables < RuntimeError; end
 
 begin
+  require "stackprof" if ENV["HOMEBREW_STACKPROF"]
   require_relative "global"
+  if ENV["HOMEBREW_STACKPROF"]
+    Homebrew.setup_gem_environment!
```

Comment:
This needs to be run before the `require "stackprof"` or it will fail if it's not installed.